service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -1,31 +1,163 @@
 package bio.terra.cbas.controllers;
 
+import static bio.terra.cbas.common.MetricsUtil.increaseEventCounter;
+
 import bio.terra.cbas.api.MethodsApi;
 import bio.terra.cbas.common.DateUtils;
 import bio.terra.cbas.dao.MethodDao;
 import bio.terra.cbas.dao.MethodVersionDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.dependencies.wes.CromwellService;
 import bio.terra.cbas.model.MethodDetails;
 import bio.terra.cbas.model.MethodLastRunDetails;
 import bio.terra.cbas.model.MethodListResponse;
 import bio.terra.cbas.model.MethodVersionDetails;
+import bio.terra.cbas.model.PostMethodRequest;
+import bio.terra.cbas.model.PostMethodResponse;
+import bio.terra.cbas.models.CbasRunSetStatus;
 import bio.terra.cbas.models.Method;
 import bio.terra.cbas.models.MethodVersion;
+import bio.terra.cbas.models.RunSet;
+import cromwell.client.model.WorkflowDescription;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
 import java.util.UUID;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 
 @Controller
 public class MethodsApiController implements MethodsApi {
+  private final CromwellService cromwellService;
   private final MethodDao methodDao;
   private final MethodVersionDao methodVersionDao;
+  private final RunSetDao runSetDao;
+
+  private static final List<String> SUPPORTED_SOURCES = new ArrayList<>(List.of("GitHub"));
```

Comment:
Could this field be an `enum` in the openapi spec?

Reply:
Good point. I will make it an enum. So if it contains any other value the field will be `null` and then we can only check for that right? 🤔